cmd/crank/beta/validate/unknown_fields.go

```diff
@@ -0,0 +1,25 @@
+package validate
+
+import (
+	"fmt"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/schema/pruning"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"strings"
+)
+
+// validateUnknownFields Validates the resource's unknown fields against the given schema and returns a list of errors.
+func validateUnknownFields(mr map[string]interface{}, sch *schema.Structural) field.ErrorList {
+	opts := schema.UnknownFieldPathOptions{
+		TrackUnknownFieldPaths: true, // to get the list of pruned unknown fields
+	}
+	errs := field.ErrorList{}
+
+	unkFlds := pruning.PruneWithOptions(mr, sch, true, opts)
+	for _, f := range unkFlds {
+		strPath := strings.Split(f, ".")
+		child := strPath[len(strPath)-1]
```

Comment:
is there any risk of `len(strPath)` being 0 and therefore this index is out of range?

Reply:
[`PruneWithOptions doc`](https://pkg.go.dev/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/pruning@v0.30.0#PruneWithOptions) says that `it returns the set of fields it prunes if TrackUnknownFieldPaths is true`. `len(strPath)` being 0 can only happen when we prune a field which does not have a path in the resource yaml. I couldn't find or think of a situation where `len(strPath)` being 0 is the case so used this approach for parsing the string and getting the unknown field name. I checked cases like base fields `metadata`, `apiVersion` or `kind` being misspelled and it all worked fine.

Reply:
good points! yep, the `strings.Split` call will never return an empty slice given our usage here (with a non empty separator) 

From https://pkg.go.dev/strings#Split:
> If s does not contain sep and sep is not empty, Split returns a slice of length 1 whose only element is s.

So this should be safe ✅